lib/Dancer2/Core/Request.pm

```diff
@@ -653,6 +653,14 @@ sub _shallow_clone {
     return $new_request;
 }
 
+
+sub route {
+    my ($self, $route) = @_;
+    if ($route) { $self->{route_matched} = $route; }
+
+    return $self->{route_matched};
```

Comment:
The concept of state is generally useful, even outside Moo.

However, I played with it now and I don't really see how to make this not-rw, since it needs to be set *after* the request. I would still prefer `set_route` instead of `route` as rw method, and `matched_route` as an entry. Mine: https://github.com/PerlDancer/Dancer2/pull/1371